strawberryfields/apps/data/feature.py

```diff
@@ -0,0 +1,148 @@
+
+
+class FeatureDataset(ABC):
+    """Base class for loading datasets of pre-calculated GBS feature vectors.
+
+        For each dataset, feature vectors, information on what orbits/events are used to construct
+        these feature vectors and the corresponding adjacency matrices are provided.
+
+    Attributes:
+        n_mean (float): mean number of photons used in feature vector calculation
+        threshold (bool): flag to indicate whether feature vectors are calculated with threshold
+            detection (i.e., detectors of zero or some photons) or with photon-number-resolving
+            detectors.
+        method (str): method used to calculate the feature vectors; "exact" for exact calculation
+         or "MC" for Monte Carlo estimation
+            estimation.
+        num_fv (int): number of feature vectors provided in the dataset
+        dim_fv (int): dimension of all feature vectors
+        featuresData (array): array of feature vectors
+        unitData (array): array of orbits/events used to construct the feature vectors
+        matData (array): array of adjacency matrices of graphs for which feature vectors
+            were calculated
+    """
+
+    @property
+    @abstractmethod
+    def _data_filename(self) -> str:
+        """Base name of files containing the data stored in the ``./FeatureData/`` directory.
+
+        For each dataset, feature vectors, information on what orbits/events are used to construct
+        these feature vectors and the corresponding adjacency matrices are provided in three
+        separate files.
+
+        Array of feature vectors should be provided in a text file. For ``_data_filename = "example"``,
+        feature vectors should be stored in ``./FeatureData/example_fv.txt``.
+
+        Array of corresponding adjacency matrices should be provided as a ``scipy.sparse.csr_matrix``
+        saved in ``.npz`` format. For ``_data_filename = "example"``, adjacency matrices
+        are stored in ``./FeatureData/example_mat.npz``."""
+        pass
+
+    @property
+    @abstractmethod
+    def _unit(self) -> str:
+        """Signifies the unit of construction of feature vectors; "orbits" or "events". This is
+        used with ``_data_filename`` to retrieve data stored in the ``./FeatureData/`` directory.
+
+        If ``_unit = "orbits"``, list of orbits used should be provided in a text file. Hence, for
+        ``_data_filename = "example"``, list of orbits should be stored in
+        ``./FeatureData/example_orbits.txt``.
+
+        If ``_unit = "events"``, list of events used should be provided in a text file. Hence, for
+        ``_data_filename = "example"``, list of events should be stored in
+        ``./FeatureData/example_events.txt``."""
+        pass
+
+    def __init__(self):
+        self.featuresData = np.loadtxt(DATA_PATH + self._data_filename + "_fv.txt")
+        self.unitData = np.loadtxt(DATA_PATH + self._data_filename + self._unit + ".txt")

```

Comment:
To save the complication of having an extra file, how about we save all the feature vectors and unitdata in one file?
e.g. save `all_data = (features, units)` in `npy` format.

Reply:
so in this file, will the orbit arrays come at the end/start of all feature vectors arrays? mentioning all orbits with every feature vector will be quite repetitive. another thing we can do is to just give it as an array in the object classes; without loading it from a file as this is constant and known for the different datasets. I personally think having a separate small file does not hurt.

Reply:
Continuing from the comment above (https://github.com/XanaduAI/strawberryfields/pull/401#discussion_r435868542), all I'm imagining is having something like:
```python
np.savez("file.npz", features=features, units=units)
```
Where we can have a single units array.